### PR TITLE
Create Article: Big-O Notation

### PR DESCRIPTION
**Pull request by angregor**
Part of Issue #1048

### CODE REVIEW

File: Algorithms-Big-O-Notation.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Big-O Notation
+
+In mathematics, Big-O notation is a symbolism used to describe and compare the limiting behavior of a function.
```
Comment (alayek) [inline Algorithms-Big-O-Notation.md]: > In mathematics, Big-O notation is a symbolism used to describe and compare the limiting behavior of a function.

Great opening, but maybe you would add a line or two explaining what you mean by _limiting behavior_.

File: Algorithms-Big-O-Notation.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+In short Big-O notation is used to describe the growth or decline of a function, usually with respect to another function.
```
Comment (alayek) [inline Algorithms-Big-O-Notation.md]:
```diff
- In short Big-O notation
+ In short, big-O notation
```

File: Algorithms-Big-O-Notation.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+For example we say that x = O(x^2) for all x > 1 or in other words, x^2 is an upper bound on x and therefore it grows faster.
```
Comment (alayek) [inline Algorithms-Big-O-Notation.md]: You might want to toss in a line describing what you mean by the `^` symbol. In this case, it is exponentiation, but in  few common programming languages, its XOR.

File: Algorithms-Big-O-Notation.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+A function's limiting behavior is how the function acts as it trends towards a particular value and in big-O notation it is usually as it trends towards infinity.
+In short, big-O notation is used to describe the growth or decline of a function, usually with respect to another function.
+
+[NOTE: x^2 is equivalent to x * x or 'x-squared']
+
+For example we say that x = O(x^2) for all x > 1 or in other words, x^2 is an upper bound on x and therefore it grows faster.
+The symbol of a claim like x = O(x^2) for all x > *n* can be substituted with  x <= x^2 for all x > *n* where *n* is the minimum number that satisfies the claim, in this case 1.
+Effectively, we say that a function f(x) that is O(g(x)) grows slower than g(x) does.
+
+Comparitively, in computer science and software development we can use big-O notation in order to describe the time complexity or efficiency of our algorithm.
```
Comment (atjonathan) [inline Algorithms-Big-O-Notation.md]: ~~of our algorithm.~~ of algorithms.

**Comment by atjonathan** [comment]
### **LGTM :+1:**

**Comment by Rafase282** [comment]
# LGTM :+1: 

Hopefully we can expand this later on. I personally would like to learn how to determine performance and all. This is one topic I passed and still learnt nothing in college.

File: Algorithms-Big-O-Notation.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+Examining the worst case scenario where the array is in reverse order, the first iteration will make *n* comparisons while the next will have to make *n* - 1 comparisons and so on until only 1 comparison must be made.
+The big-O notation for this case is therefore *n* * [(*n* - 1) / 2] which = 0.5*n*^2 - 0.5*n* = O(*n*^2) as the *n*^2 term dominates the function which allows us to ignore the other term in the function.
+
+We can confirm this analysis using [this handy big-O cheat sheet](http://bigocheatsheet.com/) that features the big-O time complexity of many commonly used data structures and algorithms
+
+It is very apparent that while for small use cases this time complexity might be alright, at a large scale bubble sort is simply not a good solution for sorting.
+This is the power of big-O notation: it allows developers to easily see the potential bottlenecks of their application, and take steps to make these more scalable.
+
+For more information on why big-O notation and algorithm analysis is important visit this [hike](https://www.freecodecamp.com/videos/big-o-notation-what-it-is-and-why-you-should-care)!
```
Comment (raisedadead) [inline Algorithms-Big-O-Notation.md]:
```diff
-[hike]
+[video challenge]
```

**Comment by raisedadead** [comment]
Everything else LGTM.

**Comment by atjonathan** [comment]
@koustuvsinha, anything you wanna add?

**Pull request merged**